app/src/main/java/edu/cascadia/mobile/apps/movies/EditMovie.java

```diff
@@ -23,28 +24,40 @@ protected void onCreate(Bundle savedInstanceState) {
         setSupportActionBar(toolbar);
 
         mDatabase = movieDatabase.getInstance(this);
+        final Intent EditIntent = this.getIntent();
+        int movieId = EditIntent.getIntExtra("movieIndex", 0);
+        //get current movie from database (dao) using movieId
+        MovieEntity currentMovie = mDatabase.movieDao().getMovie(movieId);
+        //get text fields from layout for title, director, year, runtime
+        EditText title = findViewById(R.id.edit_title);
+        EditText director = findViewById(R.id.edit_director);
+        EditText year = findViewById(R.id.edit_year);
+        EditText runtime = findViewById(R.id.edit_runtime);
+        //set title, director, year, runtime using current movie values
+        title.setText(currentMovie.getTitle());
+        director.setText(currentMovie.getDirector());
+        year.setText(currentMovie.getYear());
+        //runtime.setText(currentMovie.getRunTime());
 
-        FloatingActionButton fab = findViewById(R.id.fab);
-        fab.setOnClickListener(new View.OnClickListener() {
+        FloatingActionButton saveChanges = findViewById(R.id.saveChanges);
+        saveChanges.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                //TODO manage movie update by setting movie id if avaliable
-                int movieId = 0;
-
                 //Add Movie
                 EditText eTitle = findViewById(R.id.edit_title);
                 EditText eDirector = findViewById(R.id.edit_director);
                 EditText eYear = findViewById(R.id.edit_year);
                 EditText eRuntime = findViewById(R.id.edit_runtime);
                 int runtime = Integer.parseInt(eRuntime.getText().toString());
+                int movieId = EditIntent.getIntExtra("movieIndex", 0);
```

Comment:
Use a breakpoint here to watch the value of your movieId. This should help you debug the issue